### ISSUE TITLE
typeorm-ts-node-esm doesn't work on linux environments

### ISSUE THREAD

**Issue by Spiderbyte12** [typeorm/typeorm#8855]
## Issue Description

typeorm-ts-node-esm doesn't work on linux environments, due to env shebang with multiple arguments

### Expected Behavior

Running npx typeorm-ts-node-esm should work on linux.

### Actual Behavior

npx typeorm-ts-node-esm returns:

```
'node --loader ts-node/esm --no-warnings' no such file exists
```
This is a result of the shebang being parsed incorrectly, a known issue with env on almost all linux distros.
Running into this issue trying to run migrations from inside a docker container (on a CI pipeline)

### Steps to Reproduce

1. On linux, install typeorm
2. Run npx typeorm init.
3. Run npx typeorm-ts-node-esm

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  Linux Alpine       |
| Node.js version     | 16.14.2  | 
| Typescript version  | 4.4.4  | 
| TypeORM version     | 0.3.4 |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✅ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

Fixing this has the potential issue of breaking compatibility in other places. It's possible that the best way to approach this is to but an alternate path in to work around the shebang, and still correctly set the environment variables.

**Comment by AlexMesser** [comment]
fixed in #8821 and released in `0.3.5`. Please update your TypeORM to the latest version and reopen this issue if the error persists.

**Comment by Spiderbyte12** [comment]
Oh! I had only just installed and hadn't realised there was already a patch!
Thanks heaps, nice work.
